libraries/tools/kotlin-main-kts/src/org/jetbrains/kotlin/mainKts/scriptDef.kt

```diff
@@ -40,15 +40,18 @@ abstract class MainKtsScript(val args: Array<String>)
 const val COMPILED_SCRIPTS_CACHE_DIR_ENV_VAR = "KOTLIN_MAIN_KTS_COMPILED_SCRIPTS_CACHE_DIR"
 const val COMPILED_SCRIPTS_CACHE_DIR_PROPERTY = "kotlin.main.kts.compiled.scripts.cache.dir"
 const val COMPILED_SCRIPTS_CACHE_VERSION = 1
+const val SCRIPT_FILE_LOCATION_DEFAULT_VARIABLE_NAME = "__FILE__"
 
 class MainKtsScriptDefinition : ScriptCompilationConfiguration(
     {
-        defaultImports(DependsOn::class, Repository::class, Import::class, CompilerOptions::class)
+        defaultImports(DependsOn::class, Repository::class, Import::class, CompilerOptions::class, ScriptFileLocation::class)
         jvm {
             dependenciesFromClassContext(MainKtsScriptDefinition::class, "kotlin-main-kts", "kotlin-stdlib", "kotlin-reflect")
         }
         refineConfiguration {
+            beforeCompiling(::configureScriptFileLocationPathVariablesForCompilation)
```

Comment:
very minor: `beforeCompiling` callbacks are called after `onAnnotations` ones, so it will look a bit cleaner if you move this line to the position after the appropriate `onAnnotations` one.

Reply:
Done